extensions-core/kafka-extraction-namespace/src/main/java/io/druid/query/lookup/KafkaLookupExtractorFactory.java

```diff
@@ -268,7 +268,8 @@ public void onFailure(Throwable t)
         }
       }
       catch (InterruptedException | ExecutionException | TimeoutException e) {
-        if (!future.isDone() && !future.cancel(true) && !future.isDone()) {
+        executorService.shutdown();
+        if (!future.isDone() && !future.cancel(false)) {
```

Comment:
I could, yes. Rolling this back to `!future.isDone() && !future.cancel(true) && !future.isDone()` would be very easy, but I doubt that it's the right thing to do. To be honest, the whole class requires a heavy refactoring. I actually did that in another pull request (https://github.com/druid-io/druid/pull/3351), but that was too much of a change for a single bugfix. I would much rather push through with the refactoring.